File: backend/ruby_foundations/exercises/small_problems_101/easy_5/alphabetical_numbers.rb
NUMBERS = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten', 'eleven', 
          'twelve', 'thirteen', 'fourteen', 'fifteen', 'sixteen', 'seventeen', 'eighteen', 'nineteen']

def alphabetic_number_sort(array)
  sorted_numbers = NUMBERS.sort
  array.map do |number|
    NUMBERS.index(sorted_numbers[number])
  end
end

puts alphabetic_number_sort((0..19).to_a) == [
  8, 18, 11, 15, 5, 4, 14, 9, 19, 1, 7, 17,
  6, 16, 10, 13, 3, 12, 2, 0
]

File: frontend/javascript/exercises/list_processing/alphabetical_numbers.js

function alphabeticNumberSort(numbers) {
  var LOOKUP_TABLE = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 
  'nine', 'ten', 'eleven', 'twelve', 'thirteen', 'fourteen', 'fifteen', 
  'sixteen', 'seventeen', 'eighteen', 'nineteen'];
  var alphabeticalNumbers = numbers.map(function (number, index) {
    return LOOKUP_TABLE[index];
  });
  return alphabeticalNumbers.sort().map(function (number, index) {
    return numbers[LOOKUP_TABLE.indexOf(number)];
  });
}


console.log(alphabeticNumberSort([0, 1, 2])); // [1, 2, 0]
console.log(alphabeticNumberSort([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19])); 
// [8, 18, 11, 15, 5, 4, 14, 9, 19, 1, 7, 17, 6, 16, 10, 13, 3, 12, 2, 0]